plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java

```diff
@@ -244,10 +252,9 @@ public long getRequestCount()
     }
 
     @Override
-    public boolean isCached(Location location)
+    public boolean isCached(Location location, SchemaTableName schemaTableName)
     {
-        ValueHolder cached = cache.getIfPresent(location);
-        return cached != null && cached.getFiles().isPresent();
+        return cache.asMap().containsKey(new CacheKey(location, schemaTableName));
```

Comment:
The previous check was ensuring that the cache entry also had files materialized, which I think we want to preserve here. Basically, it's not sufficient to say that the location is cached just because there's an entry- the entry has to have completed a file listing in cache